contracts/modules/v2/StakingPool.sol

```diff
@@ -94,17 +98,48 @@ abstract contract StakingPool is IStakingPool, ERC721 {
   uint constant REWARDS_DENOMINATOR = 100;
   uint constant WEIGHT_DENOMINATOR = 100;
 
+  // product params flags
+  uint constant FLAG_PRODUCT_WEIGHT = 1;
```

Comment:
I was thinking to unify the `setWeights` and `setTargetPrice` in a single function with this signature:  
`function setProductDetails(ProductParams[] memory params)`.

The `ProductParams` contains both the weight and the target price and the uint flags would signal what updates that are intended to be applied:
```
struct ProductParams {
  uint productId;
  uint weight;
  uint targetPrice;
  uint flags; // for example: FLAG_PRODUCT_WEIGHT & FLAG_PRODUCT_PRICE
}
```

Reply:
FYI: I'm not sure if this will be the final design, still pondering if it's worth or not. I think it will become clear only after I'll work on storage packing because then we'll know if we're writing to the same slot or not or maybe something else will come up.

From the user point of view (pool manager), might be worth having them in a single function so he'd be able to update both in a single tx.